src/coreclr/scripts/jitutil.py

```diff
@@ -267,7 +282,12 @@ def safe_char(c):
             return c
         else:
             return "_"
-    return "".join(safe_char(c) for c in s)
+    # Typically, a max filename length is 256, but let's limit it far below that, because callers typically add additional
+    # strings to this.
+    max_allowed_file_name_length = 150
+    s = "".join(safe_char(c) for c in s)
```

Comment:
```suggestion
    s = "".join(safe_char(c) for c in s)[-max_allowed_file_name_length:]
```

Reply:
I'm going to leave this as-is so people (like me) not too knowledgeable about Python precedence rules can understand it more easily...